lib/stoplight/domain/light/configuration_builder_interface.rb

```diff
@@ -15,9 +15,22 @@ module ConfigurationBuilderInterface
         #
         # @param data_store [DataStore::Base]
         # @return [Stoplight::Light]
-        # @deprecated consider using +Light#with+ for reconfiguration
+        # @deprecated
         def with_data_store(data_store)
-          with(data_store:)
+          deprecate(<<~MSG)
+            Light#with_data_store is deprecated and will be removed in v6.0.0.
+
+            Circuit breakers should be configured once at creation, not cloned with
+            modifications.
+
+            Instead of:
+              light = Stoplight('api-call')
+              modified = light.with_data_store(data_stare)
+
+            Configure correctly from the start:
+              Stoplight('api-call', data_store:)
+          MSG
+          with_without_warning(data_store:)
```

Comment:
Since `#with` uses `#with_without_warning` itself, maybe we should keep `#with(data_store:)` here? 🤔